WF-712: Updated repo for new toolbelt changes (#15)
spatialos/FlexibleProjectExample: README.md

## How do I use it?
The custom layout is enabled by the `spatial` tool detecting a [new-format spatialos.json](docs/reference/project-configuration.md) file. 

> To support the FPL, we have introduced three toolbelt commands: `spatial alpha local launch`, `spatial alpha cloud upload` and `spatial alpha cloud launch`. Those commands only work for FPL projects. In order to use them, you need to:

Reviewer: ```suggestion
To support projects using FPL, we have introduced three spatial commands under `alpha` subcommand: `spatial alpha local launch`, `spatial alpha cloud upload` and `spatial alpha cloud launch`. These commands only work for FPL projects. In order to use them, you need to:
```

Reviewer 2: +1 to @angelahcliu's suggestion of `toolbelt` -> `spatial` (but use "spatial CLI" in full for clarity).

"under the `alpha` subcommand" also sounds good to me, if "subcommand" is a standard phrase. 

Two suggestions:
1. I'd start with "As part of the FPL..." rather than "To support..."
2. Do we need the sentence "These commands only work for FPL projects"? Would it not be clear already, from the first sentence (especially if you use my update) and also because you're in the FPL repo?

Author: Rephrased the sentence.

* Client: `OtherWorkers/Interactive/client/bin`

## Cleaning the project

> Run `./clean.sh` to delete all build files, including worker binaries and any intermediate files generated during the build process. Alternatively, run `git clean -xdf`.

Reviewer 2: What does running `git clean -xdf` do differently? Is this something we should explain here?

Author: This is a common git command that you can google when in doubt.

Reviewer 3: what about adding a sentence explaining what `git clean -xdf` does here? if users created new files (e.g. new types of workers) and follow the instruction without knowing what it actually does, they might be deleting useful files too and these files are gone because `git clean -xdf` cleans all untracked files, not only all build files.

## Running the project locally

To launch a local instance of SpatialOS running the project, run
```bash
$ spatial alpha local launch
```
from the SpatialOS directory (or from any location by adding the `--main_config=\<path to spatialos.json\>` flag). This starts SpatialOS locally and runs the server workers `HelloWorker` and `DiceWorker`.

> You can optionally set the `--launch_config` flag to specify a filepath to the [launch configuration](docs/reference/launch-configuration.md) of your deployment. If the flag is not set, the spatial CLI will use launch configuration specified in the `launch_config` field of your [project configuration](docs/reference/project-configuration.md) as a fall back.

Reviewer 2: ```suggestion
You can optionally set the `--launch_config` flag to specify a filepath to your deployment's [launch configuration file](docs/reference/launch-configuration.md). If the flag is not set, the spatial CLI uses the launch configuration file specified in the `launch_config` field of your [project configuration file](docs/reference/project-configuration.md).
```

Author: To be discussed in person.

> As soon as your deployment is running, you can connect game clients to it. You can find the client binaries in `OtherWorkers/Interactive/client/bin/x64/ReleaseWindows` (or `ReleaseMacOS` for Mac).

Reviewer 2: ```suggestion
As soon as your deployment is running, you can connect client-workers to it. You can find the client-worker binaries in `OtherWorkers/Interactive/client/bin/x64/ReleaseWindows` (or `ReleaseMacOS` for Mac).
```

Assuming that "game client" == "client-worker" here, and that there's no reason why we'd specifically want to say 'game client".

Author: Done. Yes, he terms are synonymous.

> Connect your client by opening a second terminal to run the binary directly (from inside the `ReleaseWindows` or `ReleaseMacOS` directories):

Reviewer 2: ```suggestion
Connect your client-worker by opening a second terminal to run the binary directly (from inside the `ReleaseWindows` or `ReleaseMacOS` directories):
```

Author: Done.

* Windows: `./Client.exe localhost 7777 <client_id>`
* macOS: `mono --arch=64 Client.exe localhost 7777 <client_id>`

> This connects a client that pings the `HelloWorker` and `DiceWorker` every few seconds and then prints the response.

Reviewer 2: ```suggestion
This connects a client-worker that pings the `HelloWorker` and `DiceWorker` every few seconds and then prints the response.
```

Author: Done.

### Logs location
The logs are stored in `logs` subdirectory of the location where your `spatialos.json` file is. For example, if my `spatialos.json` file is in `/workspace/project/SpatialOS`, the SpatialOS logs will go to `/workspace/project/SpatialOS/logs`.

## Running the project in the cloud

> Upload your project assembly to the cloud:

Reviewer: ```suggestion
To upload project assembly to the cloud:
```

Author: Done.

```bash
$ spatial alpha cloud upload -a <your-assembly-name>
```

> Start a deployment in the cloud

Reviewer: ```suggestion
To start a deployment in the cloud:
```

Author: Done.

```bash
$ spatial alpha cloud launch -d <your-deployment-name> -a <your-assembly-name>
```

## Known Issues

> * (There are no known issues at the moment.)

Reviewer 2: ```suggestion
There are no known issues at the moment.
```

Author: Done.

## Reference documentation

> [Project configuration](docs/reference/project-configuration.md)

Reviewer 2: How would you feel about adding "file" to the end of all of these? I think I'm naturally wanting to do that because of the potential for confusion (with the current project layout) between the launch configuration _file_ and the launch configuration _section_ of the worker configuration _file_. I know in the FPL there are certain similar things (e.g. there's a launchConfig _field_ inside one of the files). 

Do you reckon it's worth adding "file" wherever we're talking about a file? Or is that overkill in your opinion?

Author: Let's just add `file` across the board. Screw it.
